### PR TITLE
Add gcp capi images

### PR DESCRIPTION
**Pull request by eratnch**
Added gcp ubuntu images
Added gcp image builder documentation

### CODE REVIEW

File: images/capi/packer/gce/packer.json [hunk 1]
```diff
@@ -35,34 +35,35 @@
...
     }
   ],
   "provisioners": [
     {
       "type": "shell",
+      "environment_vars": [
+        "BUILD_NAME={{user `build_name`}}"
+      ],
       "inline": [
+        "if [ $BUILD_NAME != \"ubuntu-1804\" ]; then exit 0; fi",
```
Comment (kkeshavamurthy) [inline images/capi/packer/gce/packer.json:64]: Just curious why this was being run only for Ubuntu-18.04.

Comment (eratnch) [reply images/capi/packer/gce/packer.json]: Updated, thanks

File: images/capi/packer/gce/packer.json [hunk 2]
```diff
@@ -35,34 +35,35 @@
...
-      "image_name": "cluster-api-ubuntu-1804-{{user `kubernetes_semver` | clean_resource_name}}-{{user `build_timestamp`}}",
-      "image_family": "capi-ubuntu-1804-k8s-{{user `kubernetes_series` | clean_resource_name}}",
+      "image_name": "cluster-api-{{user `build_name`}}-{{user `kubernetes_semver` | clean_resource_name}}-{{user `build_timestamp`}}",
+      "image_family": "capi-{{user `build_name`}}-k8s-{{user `kubernetes_series` | clean_resource_name}}",
       "zone": "{{ user `zone` }}",
       "ssh_username": "ubuntu",
       "labels": {
         "build_timestamp": "{{user `build_timestamp`}}",
-        "source_ami": "{{user `ubuntu_18_04_ami`}}",
+        "source_ami": "{{user `source_ami`}}",
```
Comment (detiber) [inline images/capi/packer/gce/packer.json:49]: Should this be renamed, since AMI doesn't really have any meaning in GCP land?

Comment (eratnch) [reply images/capi/packer/gce/packer.json]: Thanks @detiber. I have removed the `source_ami` from the config json files. It works fine.

**Review by detiber — COMMENTED** [review]
/lgtm
/ok-to-test

**Comment by eratnch** [comment]
@detiber @CecileRobertMichon , can you please help get this reviewed and merged if ok.
Thanks in advance.

File: images/capi/Makefile [hunk 3]
```diff
@@ -180,9 +180,9 @@ HAPROXY_OVA_ESX_BUILD_NAMES			:=	$(addprefix haproxy-ova-esx-,$(PHOTON_VERSIONS)
 HAPROXY_OVA_VSPHERE_BUILD_NAMES		:=	$(addprefix haproxy-ova-vsphere-,$(PHOTON_VERSIONS))
 
 AMI_BUILD_NAMES			?=	ami-centos-7 ami-ubuntu-1804 ami-ubuntu-2004 ami-amazon-2
-GCE_BUILD_NAMES			?=	gce-default
-AZURE_BUILD_VHD_NAMES		?=	azure-vhd-ubuntu-1804 azure-vhd-centos-7 ## azure-vhd-ubuntu-2004
-AZURE_BUILD_SIG_NAMES		?=	azure-sig-ubuntu-1804 azure-sig-centos-7 ## azure-sig-ubuntu-2004
+GCE_BUILD_NAMES			?=	gce-ubuntu-1804 gce-ubuntu-2004
+AZURE_BUILD_VHD_NAMES		?=	azure-vhd-ubuntu-1804 azure-vhd-ubuntu-2004 azure-vhd-centos-7
```
Comment (CecileRobertMichon) [inline images/capi/Makefile:184]: please keep azure 2004 disabled for now, see #429

File: images/capi/Makefile [hunk 4]
```diff
@@ -180,9 +180,9 @@ HAPROXY_OVA_ESX_BUILD_NAMES			:=	$(addprefix haproxy-ova-esx-,$(PHOTON_VERSIONS)
...
+AZURE_BUILD_SIG_NAMES		?=	azure-sig-ubuntu-1804 azure-sig-ubuntu-2004 azure-sig-centos-7
```
Comment (CecileRobertMichon) [inline images/capi/Makefile:185]: same here

File: images/capi/packer/gce/packer.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
     "ansible_extra_vars": "",
     "project_id": "{{env `GCP_PROJECT_ID`}}",
     "build_timestamp": "{{timestamp}}",
-    "source_image_family": "ubuntu-1804-lts",
+    "source_image_family": "{{user `build_name`}}-lts",
```
Comment (CecileRobertMichon) [inline images/capi/packer/gce/packer.json:7]: is there an issue tracking adding CI for GCP?

**Comment by CecileRobertMichon** [comment]
/test pull-azure-vhds

this should fail now because of 20.04 ^

File: images/capi/packer/gce/ubuntu-2004.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+    "build_name": "ubuntu-2004",
```
Comment (CecileRobertMichon) [inline images/capi/packer/gce/ubuntu-2004.json:2]: heads up you might run into https://github.com/kubernetes-sigs/image-builder/issues/429#issuecomment-730054075 with 20.04. We've found that it's not Azure specific, it's an issue with upgrade of a particular version of systemd. We are working with Canonical to address this.

File: images/capi/Makefile (hunk 3 above)
Comment (eratnch) [reply images/capi/Makefile:184]: thanks @CecileRobertMichon for the context, will do.

Comment (eratnch) [reply images/capi/Makefile:184]: Done

File: images/capi/Makefile (hunk 4 above)
Comment (eratnch) [reply images/capi/Makefile:185]: Done

File: images/capi/packer/gce/packer.json (hunk 5 above)
Comment (eratnch) [reply images/capi/packer/gce/packer.json:7]: Created #437.

File: images/capi/packer/gce/ubuntu-2004.json (hunk 6 above)
Comment (eratnch) [reply images/capi/packer/gce/ubuntu-2004.json:2]: Will re-enable ubuntu 20.04 for GCP once #429 is resolved.

File: images/capi/Makefile (hunk 3 above)
Comment (CecileRobertMichon) [reply images/capi/Makefile:184]: I still see `azure-vhd-ubuntu-2004`, did you push the update?

Comment (eratnch) [reply images/capi/Makefile]: I did now, can you pls check?

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm
/approve

**Pull request merged**